web-admin/src/routes/[organization]/[project]/explore/[dashboard]/+layout.ts

```diff
@@ -11,31 +13,57 @@ import {
 export const load = async ({ params, depends, parent }) => {
   const { project, runtime } = await parent();
 
-  const exploreName = params.dashboard;
+  const { dashboard: exploreName } = params;
 
   depends(exploreName, "explore");
 
+  let explore: V1Resource | undefined;
+  let metricsView: V1Resource | undefined;
+  let defaultExplorePreset: V1ExplorePreset | undefined;
+  let exploreStateFromYAMLConfig: Partial<MetricsExplorerEntity> = {};
+  let initExploreState: Partial<MetricsExplorerEntity> | undefined = undefined;
   try {
-    const { explore, metricsView, defaultExplorePreset } =
-      await fetchExploreSpec(runtime?.instanceId, exploreName);
-
-    // used to merge home bookmark to url state
-    const bookmarks = await fetchBookmarks(project.id, exploreName);
-
-    return {
+    ({
       explore,
       metricsView,
       defaultExplorePreset,
-      homeBookmark: bookmarks.find(isHomeBookmark),
-    };
+      exploreStateFromYAMLConfig,
+    } = await fetchExploreSpec(runtime?.instanceId, exploreName));
   } catch {
     // error handled in +page.svelte for now
     // TODO: move it here
     return {
       explore: <V1Resource>{},
       metricsView: <V1Resource>{},
       defaultExplorePreset: <V1ExplorePreset>{},
-      homeBookmark: undefined,
+      exploreStateFromYAMLConfig,
     };
   }
+
+  const metricsViewSpec = metricsView.metricsView?.state?.validSpec ?? {};
+  const exploreSpec = explore.explore?.state?.validSpec ?? {};
+
+  try {
+    const bookmarks = await fetchBookmarks(project.id, exploreName);
+    const homeBookmark = bookmarks.find(isHomeBookmark);
+
+    if (homeBookmark) {
+      initExploreState = getDashboardStateFromUrl(
```

Comment:
Let's name this state `bookmarkExploreState` and pass it down with that name until we merge it with the other state sources